src/en/kmanga/src/eu/kanade/tachiyomi/extension/en/kmanga/KManga.kt

```diff
@@ -0,0 +1,369 @@
+            thumbnail_url = resolve(titleDetailsObject["banner_image_url"]!!).jsonPrimitive.content
+            val genreIdRefs = resolve(titleDetailsObject["genre_id_list"]!!).jsonArray
+            val genreIds = genreIdRefs.map { resolve(it).jsonPrimitive.content.toInt() }
+            genre = genreIds.mapNotNull { genreMap[it] }.joinToString()
+            status = SManga.ONGOING
+        }
+    }
+
+    // Chapters
+    override fun chapterListRequest(manga: SManga): Request {
+        return GET(baseUrl + manga.url, headers)
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val document = response.asJsoup()
+        val nuxtData = document.selectFirst("script#__NUXT_DATA__")?.data()
+            ?: throw Exception("Could not find Nuxt data")
+
+        val rootArray = json.decodeFromString<JsonArray>(nuxtData)
+        fun resolve(ref: JsonElement): JsonElement = rootArray[ref.jsonPrimitive.content.toInt()]
+
+        val titleDetailsObject = rootArray.first { it is JsonObject && it.jsonObject.containsKey("episode_id_list") }.jsonObject
+        val episodeIdRefs = resolve(titleDetailsObject["episode_id_list"]!!).jsonArray
+        val episodeIds = episodeIdRefs.map { resolve(it).jsonPrimitive.content }
+
+        val (birthday, expires) = getBirthdayCookie(response.request.url)
+
+        val formBody = FormBody.Builder()
+            .add("episode_id_list", episodeIds.joinToString(","))
+            .build()
+
+        val params = (0 until formBody.size).associate { formBody.name(it) to formBody.value(it) }
+        val hash = generateHash(params, birthday, expires)
+
+        val postHeaders = headersBuilder()
+            .add("Accept", "application/json")
+            .add("Content-Type", "application/x-www-form-urlencoded")
+            .add("Origin", baseUrl)
+            .add("x-kmanga-is-crawler", "false")
+            .add("x-kmanga-hash", hash)
+            .build()
+
+        val apiRequest = POST("$apiUrl/episode/list", postHeaders, formBody)
+        val apiResponse = client.newCall(apiRequest).execute()
+
+        if (!apiResponse.isSuccessful) {
+            throw Exception("API request failed with code ${apiResponse.code}: ${apiResponse.body.string()}")
+        }
+
+        val result = json.decodeFromString<EpisodeListResponse>(apiResponse.body.string())
+
+        return result.episode_list.map { chapter ->
+            SChapter.create().apply {
+                url = "/title/${chapter.title_id}/episode/${chapter.episode_id}"
+                name = if (chapter.point > 0 && chapter.badge != 3 && chapter.rental_finish_time == null) {
+                    "🔒 ${chapter.episode_name}"
+                } else {
+                    chapter.episode_name
+                }
+                date_upload = runCatching {
+                    dateFormat.parse(chapter.start_time)!!.time
+                }.getOrDefault(0L)
```

Comment:
```suggestion
                date_upload = dateFormat.tryParse(chapter.start_time)
```
import from `keiyoushi.utils`